alt_e2eshark/run.py

```diff
@@ -0,0 +1,52 @@
+import sys
+from pathlib import Path
+sys.path.append(Path(__file__).parent)
+print(Path(__file__).parent)
+from e2e_testing.storage import TestTensors
+from e2e_testing.framework import *
+from e2e_testing.registry import GLOBAL_TEST_LIST
+
+test_dir = Path(__file__).parent
+
+# importing the test generating files will register them to 
+# GLOBAL_TEST_LIST
+from onnx_tests.operators import model
+
+# import frontend test configs:
+from e2e_testing.test_configs.onnxconfig import OnnxTestConfig
+# import backend
+from e2e_testing.backends import SimpleIREEBackend
+
+def main():
+    #TODO: add argparse to customize config/backend/testlist/etc.
+    config = OnnxTestConfig(str(test_dir), SimpleIREEBackend())
+    test_list = GLOBAL_TEST_LIST
+    run_tests(test_list, config, test_dir)
+
+
+#TODO: finish and move elsewhere
+
+def run_tests(test_list, config, test_dir):
+    #TODO: multi-process
+    results = []
+    for t in test_list:
+       log_dir = str(test_dir) + '/' + t.unique_name
+       print(log_dir)
+       #TODO: Add logging/log_dir
+       inst = t.model_constructor(str(test_dir)) 
+       inputs = inst.construct_inputs()
+       golden_outputs = inst.forward(inputs)
+
+       mlir_module = config.mlir_import(inst)
+       buffer = config.compile(mlir_module)
+       callable_compiled_module = config.backend.load(buffer)
+       outputs = callable_compiled_module(inputs)
+       result = TestResult(name=t.unique_name, input=inputs,gold_output=golden_outputs,output=outputs)
```

Comment:
pytest could also help here. You could use a fixture or base test class to track this logic
https://docs.pytest.org/en/stable/explanation/fixtures.html

then could use https://pypi.org/project/pytest-xdist/ for multi-threading (or https://pypi.org/project/pytest-shard/ for running across machines)

Reply:
Yeah, my original goal was to keep the core functionality (besides trying to mininmize command-line scripting the running of concatenated python files from python) of the current run.py strategy, but simply factor each piece into a more appropriate python object. I'll definitely need to coordinate with someone about pytest and multi-threadded testing, since I know there are some subtleties there that I'm likely not prepared for yet.